src/MongoDB.Driver.Core/Core/Misc/ServerVersion.cs

```diff
@@ -0,0 +1,463 @@
+        }
+
+        // static methods
+        /// <summary>
+        /// Parses a string representation of a server version.
+        /// </summary>
+        /// <param name="value">The string value to parse.</param>
+        /// <returns>A server version.</returns>
+        /// <exception cref="FormatException">value</exception>
+        public static ServerVersion Parse(string value)
+        {
+            ServerVersion result;
+            if (TryParse(value, out result))
+            {
+                return result;
+            }
+
+            throw new FormatException($"Invalid ServerVersion: '{value}'.");
+        }
+
+        /// <summary>
+        /// Tries to parse a string representation of a server version.
+        /// </summary>
+        /// <param name="value">The string value to parse.</param>
+        /// <param name="result">The result.</param>
+        /// <returns>True if the string representation was parsed successfully; otherwise false.</returns>
+        public static bool TryParse(string value, out ServerVersion result)
+        {
+            if (string.IsNullOrEmpty(value))
+            {
+                result = null;
+                return false;
+            }
+            var pattern = @"^(?<major>\d+)\.(?<minor>\d+)(\.(?<patch>\d+)(-rc(?<releaseCandidate>\d+))?(-(?<internalBuild>\d+)-g(?<commitHash>[0-9a-f]{4,40}))?)?$";
+            var match = Regex.Match(value, pattern);
+
+            if (!match.Success)
+            {
+                result = null;
+                return false;
+            }
+
+            var major = int.Parse(match.Groups["major"].Value);
+            var minor = int.Parse(match.Groups["minor"].Value);
+            var patch = match.Groups["patch"].Success ? int.Parse(match.Groups["patch"].Value) : 0;
+            var releaseCandidateEntry = match.Groups["releaseCandidate"].Success ? match.Groups["releaseCandidate"].Value : null;
+            var internalBuildEntry = match.Groups["internalBuild"].Success ? match.Groups["internalBuild"].Value : null;
+            var commitHash = match.Groups["commitHash"].Success ? match.Groups["commitHash"].Value : null;
+
+            int? releaseCandidate = null;
+            if (releaseCandidateEntry != null && int.TryParse(releaseCandidateEntry, out int releaseCandidateParsed))
+            {
+                releaseCandidate = releaseCandidateParsed;
+            }
+            int? internalBuild = null;
+            if (internalBuildEntry != null && int.TryParse(internalBuildEntry, out int internalBuildParsed))
+            {
+                internalBuild = internalBuildParsed;
+            }
+
+            result = new ServerVersion(major, minor, patch, releaseCandidate, internalBuild, commitHash);
+            return true;
+        }
+
+        // public operators
+        /// <summary>
+        /// Casts a SemanticVersion to a ServerVersion.
+        /// </summary>
+        /// <param name="semanticVersion">The SemanticVersion.</param>
```

Comment:
as far as I know, we usually don't start the words from the capital letter except for the first word

Reply:
Removed the method.